ddtrace/llmobs/_integrations/gemini.py

```diff
@@ -1,18 +1,154 @@
+        span.set_tag_str(METADATA, json.dumps(metadata, default=_unserializable_default_repr))
+
+        system_instruction = _get_attr(instance, "_system_instruction", None)
+        input_contents = get_argument_value(args, kwargs, 0, "contents")
+        input_messages = self._extract_input_message(input_contents, system_instruction)
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages, default=_unserializable_default_repr))
+
+        if span.error or generations is None:
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps([{"content": ""}]))
+        else:
+            output_messages = self._extract_output_message(generations)
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages, default=_unserializable_default_repr))
+
+        usage = self._get_llmobs_metrics_tags(span)
+        if usage:
+            span.set_tag_str(METRICS, json.dumps(usage, default=_unserializable_default_repr))
+
+    @staticmethod
+    def _llmobs_set_metadata(kwargs, instance):
+        metadata = {}
+        model_config = instance._generation_config or {}
+        request_config = kwargs.get("generation_config", {})
+        parameters = ("temperature", "max_output_tokens", "candidate_count", "top_p", "top_k")
+        for param in parameters:
+            model_config_value = _get_attr(model_config, param, None)
+            request_config_value = _get_attr(request_config, param, None)
+            if model_config_value or request_config_value:
+                metadata[param] = request_config_value or model_config_value
+        return metadata
+
+    @staticmethod
+    def _extract_message_from_part(part, role):
+        text = _get_attr(part, "text", "")
+        function_call = _get_attr(part, "function_call", None)
+        function_response = _get_attr(part, "function_response", None)
+        message = {"content": text, "role": role}
+        if function_call:
+            function_call_dict = function_call
+            if not isinstance(function_call, dict):
+                function_call_dict = type(function_call).to_dict(function_call)
+            message["tool_calls"] = [
+                {"name": function_call_dict.get("name", ""), "arguments": function_call_dict.get("args", {})}
+            ]
+        if function_response:
+            function_response_dict = function_response
+            if not isinstance(function_response, dict):
+                function_response_dict = type(function_response).to_dict(function_response)
+            message["content"] = "[tool result: {}]".format(function_response_dict.get("response", ""))
+        return message
+
+    def _extract_input_message(self, contents, system_instruction=None):
+        messages = []
+        if system_instruction:
+            for part in system_instruction.parts:
+                messages.append({"content": part.text or "", "role": "system"})
+        if isinstance(contents, str):
```

Comment:
should we be defaulting `role` to `user` here? `role` isn't required, so i think we should just not set it